File: Visual C++/Console Applications/Imap/Imap.cpp

#include "stdafx.h"

#include <comdef.h>
#include <atlbase.h>
#include <windows.h>
#include <stdio.h>

////////////////////////////////////////////////////////////////////////////////

#import "..\..\include\AxEmail.tlb"
#include "..\..\include\AxEmailConstants.h"

////////////////////////////////////////////////////////////////////////////////

LPTSTR ReadInput( LPCTSTR lptszTitle, BOOL bAllowEmpty = FALSE );

////////////////////////////////////////////////////////////////////////////////


int main(void)
{

  LONG			lLastError		= 0L;
  LONG			lNumMessages	= 0L;
  _bstr_t			bstrHostName, bstrAccount, bstrPassword;
  LPTSTR			cp;
  BOOL			bSecure			= FALSE;
  AxEmail::IImapPtr oImap = NULL;
  AxEmail::IMessagePtr oMessage = NULL;
  AxEmail::IConstantsPtr oConstants = NULL;

  // initialize COM
  CoInitialize(NULL);

  oConstants.CreateInstance(__uuidof(AxEmail::Constants));
  oImap.CreateInstance(__uuidof(AxEmail::Imap));
  if( oImap == NULL || oConstants == NULL )
  {
    _tprintf( _T("Failed to create AxEmail:Imap and AxEmail:Constants.\n") );		
    goto _EndMain;
  }

  // A license key is required to unlock this component after the trial period has expired.
  // Assign the LicenseKey property every time a new instance of this component is created (see code below). 
  // Alternatively, the LicenseKey property can be set automatically. This requires the license key to be stored in the registry.
  // For details, see manual, chapter "Product Activation".
  /*
  oImap->LicenseKey = _T("XXXXX-XXXXX-XXXXX");
  */

  // Component info
  _tprintf( _T("Auron Email Component %s\nBuild: %s\nModule: %s\nLicense Status: %s\nLicense Key: %s\n\n"), ( LPCTSTR ) oImap->Version, ( LPCTSTR ) oImap->Build, ( LPCTSTR ) oImap->Module, ( LPCTSTR ) oImap->LicenseStatus, ( LPCTSTR ) oImap->LicenseKey );

  // Log File
  // oImap->LogFile = _T("C:\\Imap.log");
  _tprintf( _T("Log File used: %s\n\n"), ( LPCTSTR ) oImap->LogFile );


  cp				= ReadInput( _T("Imap Server") );
  bSecure			= _tcsicmp( cp, _T("imap.gmail.com") ) == 0;
  bstrHostName	= cp;
  bstrAccount		= ReadInput( _T("Imap mailbox account") );
  bstrPassword	= ReadInput( _T("Imap mailbox password") );

  if( bSecure )
  {
    oImap->SetSecure( 993 );
    _tprintf( _T("SetSecure, result: %ld (%s)\n"), oImap->LastError, ( LPCTSTR ) oImap->GetErrorDescription( oImap->LastError ) ); 
  }

  // Connect
  _tprintf( _T("Connecting...\n") );
  oImap->Connect ( bstrHostName, bstrAccount, bstrPassword );
  _tprintf( _T("Connect, result: %ld (%s)\n"), oImap->LastError, ( LPCTSTR ) oImap->GetErrorDescription( oImap->LastError ) ); 
  if( oImap->LastError != 0 )
    goto _EndMain;


  oMessage = oImap->FetchFirstEmail(oConstants->IMAP_SK_ALL);
  while (oImap->LastError == 0)
  {
    _tprintf( _T("Mail ID       : %d\n"), oMessage->ImapUID );
    _tprintf( _T("  Mail From   : %s\n"), ( LPCTSTR ) oMessage->FromAddress );	
    _tprintf( _T("  Mail To     : %s\n"), ( LPCTSTR ) oMessage->ToAddress );	
    _tprintf( _T("  Mail Subject: %s\n"), ( LPCTSTR ) oMessage->Subject );	
    _tprintf( _T("  Mail Size   : %ld\n"), oMessage->Size );
    _tprintf( _T("  Mail Sent   : %s\n"), ( LPCTSTR ) oMessage->Date );

    _tprintf( _T("\n") );

    oMessage = oImap->FetchNextEmail();
  }

  oImap->Disconnect();

_EndMain:

  if( oImap != NULL )
  {
    oImap.Release();
  }

  _tprintf( _T("Ready.\n") );

  return 0;
}

////////////////////////////////////////////////////////////////////////////////

LPTSTR ReadInput( LPCTSTR lptszTitle, BOOL bAllowEmpty )
{
  static TCHAR		tszInput [ 255 + 1 ] = { _T('\0') };

  _tprintf( _T("%s:\n"), lptszTitle );
  do
  {
    _tprintf ( _T("   > ") );
    // scanf ( "%s", tszInput );
    fflush(stdin); 
    fflush(stdout); 
    _fgetts( tszInput, 255, stdin );
    if( tszInput[ 0 ] != _T('\0') && tszInput[ _tcsclen( tszInput ) - 1  ] == _T('\n') )
      tszInput[ _tcsclen( tszInput ) - 1  ] = _T('\0');
  } while( _tcsclen ( tszInput ) == 0 && ! bAllowEmpty );

  return tszInput;
}

///////////////////////////////////////////////////////////////////////////////////////////

File: PHP/imap.php
<?php
  $arrMails = array();
  $strResult = 'n/a';
  
  $objImap = new COM ("AxEmail.Imap", NULL, CP_UTF8);
      
  // A license key is required to unlock this component after the trial period has expired.
  // Assign the LicenseKey property every time a new instance of this component is created (see code below). 
  // Alternatively, the LicenseKey property can be set automatically. This requires the license key to be stored in the registry.
  // For details, see manual, chapter "Product Activation".
  //
  // $objImap->LicenseKey = "XXXXX-XXXXX-XXXXX"; 

  $objImap->LogFile = sys_get_temp_dir()."Imap.log"; 
  // Windows default: 'C:\Windows\Temp\Imap.log'
  
  if (isset($_POST['btnListMessages']))
  {
    $objEmailConstants = new COM("AxEmail.Constants", NULL, CP_UTF8);
    $objEmail = new COM("AxEmail.Message", NULL, CP_UTF8);
    
    $objImap->Authentication = $objEmailConstants->IMAP_AUTH_AUTO;
    
    if (isset($_POST['cbxSecure']))
      $objImap->SetSecure(); // 993 is the default secure IMAP port
    
    if ($objImap->LastError == 0)
    {
      $objImap->Connect($_POST['txtHost'], $_POST['txtAccount'], $_POST['txtPassword']);
    }
    
    if ($objImap->LastError == 0)
    {
      $nCnt = 0;      
      $objEmail = $objImap->FetchFirstEmailHeader($objEmailConstants->IMAP_SK_ALL);
      while ($objImap->LastError == 0)
      {
        $nCnt++;
        if ($nCnt == 100) break;
        $arrMails[] = $objEmail;
        $objEmail = $objImap->FetchNextEmailHeader();
      }
    }
        
    $strResult = $objImap->LastError . ": " . $objImap->GetErrorDescription($objImap->LastError);    
  }

// HTML-CSS layout includes, no code there!
include('css/Header.html'); 
include('css/Menu.html'); 
?>
    <div class="container">
      <h1>Auron Email Component PHP IMAP Sample</h1>
      <hr />
      <p>
      The Internet Message Access Protocol (IMAP) is a widely used protocol for accessing and maintaining a set of remote mailboxes. 
      </p>
      <form action="imap.php" method="post">
        <h2>Component:</h2>
        <h3>Build: <?php echo $objImap->Build ?>; Module: <?php echo $objImap->Module ?></h3>
      
        <!-- Host, Secure -->
        <label for="Host">Mailserver:</label>
        <p>
          <input type="text" id="Host" name="txtHost" value="imap.gmail.com" />
          
          <input type="checkbox" class="cbFix" id="Secure" name="cbxSecure" checked />
          <label for="Secure">Secure</label>
        </p>
        
        <!-- Account -->
        <label for="Account">Account:</label>
        <p>
          <input type="text" id="Account" name="txtAccount" value="user@gmail.com" />
        </p>
        
        <!-- Password -->
        <label for="Password">Password:</label>
        <p>
          <input type="password" id="Password" name="txtPassword" value="" />
        </p>
        
        <!-- Empty row -->
        <div class="clearRow"></div>
        
        <!-- List Messages button -->
        <div class="clearLabel"></div>
        <p>
          <input type="submit" name="btnListMessages" value="List First 100 Messages" />
        </p>
        
        <!-- Messages Listbox -->
        <label for="Received">Messages received:</label>
        <p>
          <select id="Received" class="FullWidth" name="lvMessages" size="10">
          <?php
          foreach($arrMails as $Mail)
          {
            echo '
            <option>'.$Mail->Date . ": " . $Mail->FromAddress . "; Subject " . $Mail->Subject.'</option>';
          }
          ?>
          </select>
        </p>
        
        <!-- Result -->
        <label for="Result" class="Bold">Result:</label>
        <p>
          <input type="text" id="Result" name="txtResult" class="FullWidth Bold" class="FullWidth" value="<?php echo $strResult; ?>" />
        </p>
      </form>  
      <p>
        This demo uses the Auron Email Component, an <a href="https://www.auronsoftware.com" target="_blank">Auron Software</a> product.<br />
        <a href="index.php">Back to main page</a> 
      </p>
    </div><!-- /container -->
<?php 
//HTML-CSS layout include, no code there!
include('css/Footer.html'); ?>